src/SAML2/Binding/ReceivedMessageQueryString.php

```diff
@@ -67,108 +67,78 @@ private function __construct($samlMessage)
     }
 
     /**
-     * @param string $query
+     * @param array $requestParams
      * @return ReceivedMessageQueryString
-     * @throws \AdactiveSas\Saml2BridgeBundle\SAML2\Binding\Exception\InvalidRequestException
-     * @throws \AdactiveSas\Saml2BridgeBundle\SAML2\Binding\Exception\InvalidReceivedMessageQueryStringException
-     *
      * @SuppressWarnings(PHPMD.CyclomaticComplexity) Extensive validation
      * @SuppressWarnings(PHPMD.NPathComplexity) Extensive validation
      */
-    public static function parse($query)
+    public static function parse(array $requestParams)
```

Comment:
I think this is confusing, the `ReceivedMessageQueryString` is intended to decrypt only the HTTP query string and now it's decrypting request data also. 

Maybe should be easier to maintain (and test) if we completely separate Query decryption and Body. I don't this the SAML allow to use QueryString data with the POST.